src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,34 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = EventBase.ScrubTenant(endpointUri),
```

Comment:
Hmm, unlike the other two conversations ([that one](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/326/files#r113039586) and [that one](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/326/files#r113039440)), here there is no specific technical reason preventing us from moving the `ScrubTenant(...)` part into the setter. So it is mostly a matter of aesthetics. I mean, if we actually change the HttpPath setter into accepting a `Uri` instance, and if this also applies to the `QueryParams` line, the code would probably become:

    {
        HttpPath = endpointUri,
        QueryParams = endpointUri
    }

So we pass same information into 2 different setters. If we went that far, would we consider combine them into one setter `{Uri = endpointUri}`, and let that setter initialize 2 underlying fields, although that way we would lose the 1:1 mapping between current setters and their fields on-the-wire?